stdlib/public/SDK/Foundation/NSStringAPI.swift

```diff
@@ -417,6 +417,20 @@ extension StringProtocol where Index == String.Index {
     return self._ephemeralString as NSString
   }
 
+  func _withCore<Result>(_ apply: (_StringCore) throws -> Result) rethrows -> Result {
+    return try apply(self._emphemerialString._core)
+  }
+
+  func _withReference<Result>(_ apply: (NSString) throws -> Result) rethrows -> Result {
+    return try _withCore { (core) throws -> Result in
```

Comment:
It doesn't seem to be necessary to me the way `_ephemeralString` is implemented now. @dabrahams and @milseman would know better.